rmw_opensplice_cpp/src/rmw_publisher.cpp

```diff
@@ -170,6 +170,13 @@ rmw_create_publisher(
 
   publisher->implementation_identifier = opensplice_cpp_identifier;
   publisher->data = publisher_info;
+  publisher->topic_name = reinterpret_cast<const char *>(
+    rmw_allocate(sizeof(char) * strlen(topic_name) + 1));
+  if (!publisher->topic_name) {
+    RMW_SET_ERROR_MSG("failed to allocate memory for node name");
+    goto fail;
+  }
+  memcpy(const_cast<char *>(publisher->topic_name), topic_name, strlen(topic_name) + 1);
```

Comment:
I don't really care what we do when copying -- we can use memcpy with or without `sizeof(char)` or use `strncpy` + `strlen`. For the `memcpy` call, I was just trying to say the `sizeof(char)` is for readability not correctness or trying to support systems where `char` is not a byte.